### ISSUE TITLE
Jinja2 no test named equalto

### ISSUE THREAD

**Issue by sylviapossenau** [ansible/ansible#57311]
##### SUMMARY
Trying to use the filter: selectattr('name','equalto',item.name) results in a TemplateRunTimeError

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Name: Jinja2
Version: 2.10.1
Summary: A small but fast and easy to use stand-alone template engine written in pure python.
Home-page: http://jinja.pocoo.org/
Author: Armin Ronacher
Author-email: armin.ronacher@active-4.com
License: BSD
Location: /usr/lib/python2.7/site-packages/jinja2

##### ANSIBLE VERSION
```
ansible 2.8.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/admin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Apr  9 2019, 14:30:50) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### CONFIGURATION
```
```

##### OS / ENVIRONMENT
CentOS 7

##### STEPS TO REPRODUCE
[Found this example on Stackoverflow](https://stackoverflow.com/questions/42673166/append-to-dict-inside-a-list-in-ansible)
Since versions may clash updated system:
```bash
$ yum update
$ pip install Jinja2
$ pip install --upgrade Jinja2
#reboot just in case

$ ansible-playbook so.yaml
```

```yaml
---
- hosts: localhost
  gather_facts: no
  vars:
    users:
      - name: foo
        database: foodb
      - name: bar
        database: bardb
    secrets:
      - name: foo
        password: foopass
      - name: bar
        password: barpass
  tasks:
    - set_fact:
        tmp_user: "{{ item | combine(secrets | selectattr('name','equalto',item.name) | first) }}"
      with_items: "{{ users }}"
      register: tmp_users
    - set_fact:
        users: "{{ tmp_users.results | map(attribute='ansible_facts.tmp_user') | list }}"
    - debug:
        msg: "{{ users }}"
```

##### EXPECTED RESULTS
I expect the tasks to not fail. 

##### ACTUAL RESULTS
```
[admin@ansiblevm ansible-test]$ ansible-playbook so.yaml -vvvv
ansible-playbook 2.8.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/admin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Apr  9 2019, 14:30:50) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
Using /etc/ansible/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
script declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
auto declined parsing /etc/ansible/hosts as it did not pass it's verify_file() method
Parsed /etc/ansible/hosts inventory source with ini plugin
Loading callback plugin default of type stdout, v2.0 from /usr/lib/python2.7/site-packages/ansible/plugins/callback/default.pyc

PLAYBOOK: so.yaml ***********************************************************************************************
Positional arguments: so.yaml
become_method: sudo
inventory: (u'/etc/ansible/hosts',)
forks: 5
tags: (u'all',)
verbosity: 4
connection: smart
timeout: 10
1 plays in so.yaml

PLAY [localhost] ************************************************************************************************
META: ran handlers

TASK [set_fact] *************************************************************************************************
task path: /home/admin/ansible-test/so.yaml:16
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 110, in run
    item_results = self._run_loop(items)
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 370, in _run_loop
    res = self._execute(variables=task_vars)
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 576, in _execute
    self._task.post_validate(templar=templar)
  File "/usr/lib/python2.7/site-packages/ansible/playbook/task.py", line 268, in post_validate
    super(Task, self).post_validate(templar)
  File "/usr/lib/python2.7/site-packages/ansible/playbook/base.py", line 382, in post_validate
    value = templar.template(getattr(self, name))
  File "/usr/lib/python2.7/site-packages/ansible/template/__init__.py", line 584, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/site-packages/ansible/template/__init__.py", line 539, in template
    disable_lookups=disable_lookups,
  File "/usr/lib/python2.7/site-packages/ansible/template/__init__.py", line 804, in do_template
    res = j2_concat(rf)
  File "<template>", line 12, in root
  File "/usr/lib/python2.7/site-packages/jinja2/filters.py", line 349, in do_first
    return next(iter(seq))
  File "/usr/lib/python2.7/site-packages/jinja2/filters.py", line 931, in _select_or_reject
    if modfunc(func(transfunc(item))):
  File "/usr/lib/python2.7/site-packages/jinja2/filters.py", line 925, in <lambda>
    name, item, args, kwargs)
  File "/usr/lib/python2.7/site-packages/jinja2/environment.py", line 438, in call_test
    raise TemplateRuntimeError('no test named %r' % name)
TemplateRuntimeError: no test named 'equalto'

fatal: [localhost]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}

PLAY RECAP ******************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0

```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/template`](https://github.com/ansible/ansible/blob/devel/lib/ansible/template)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
bot_broken

**Comment by mkrizek** [comment]
This does not seem like an Ansible issue but rather an issue with your environment:
```
$ ansible --version
ansible 2.8.0
  config file = /home/mkrizek/src/ansible/ansible.cfg
  configured module search path = ['/home/mkrizek/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/mkrizek/src/ansible/lib/ansible
  executable location = /home/mkrizek/src/ansible/venv/bin/ansible
  python version = 3.7.3 (default, May 11 2019, 00:38:04) [GCC 9.1.1 20190503 (Red Hat 9.1.1-1)]
```

```
$ pip show Jinja2
Name: Jinja2
Version: 2.10.1
Summary: A small but fast and easy to use stand-alone template engine written in pure python.
Home-page: http://jinja.pocoo.org/
Author: Armin Ronacher
Author-email: armin.ronacher@active-4.com
License: BSD
Location: /home/mkrizek/src/ansible/venv/lib/python3.7/site-packages
Requires: MarkupSafe
Required-by:
```

```
$ cat issues/57311.yml 
- hosts: localhost
  gather_facts: no
  vars:
    users:
      - name: foo
        database: foodb
      - name: bar
        database: bardb
    secrets:
      - name: foo
        password: foopass
      - name: bar
        password: barpass
  tasks:
    - set_fact:
        tmp_user: "{{ item | combine(secrets | selectattr('name','equalto',item.name) | first) }}"
      with_items: "{{ users }}"
      register: tmp_users
    - set_fact:
        users: "{{ tmp_users.results | map(attribute='ansible_facts.tmp_user') | list }}"
    - debug:
        msg: "{{ users }}"
```

```
$ ansible-playbook issues/57311.yml

PLAY [localhost] ****************************************************************************************************************************

TASK [set_fact] *****************************************************************************************************************************
ok: [localhost] => (item={'name': 'foo', 'database': 'foodb'})
ok: [localhost] => (item={'name': 'bar', 'database': 'bardb'})

TASK [set_fact] *****************************************************************************************************************************
ok: [localhost]

TASK [debug] ********************************************************************************************************************************
ok: [localhost] => {
    "msg": [
        {
            "database": "foodb",
            "name": "foo",
            "password": "foopass"
        },
        {
            "database": "bardb",
            "name": "bar",
            "password": "barpass"
        }
    ]
}

PLAY RECAP **********************************************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

IIRC `equalto` was added in Jinja2 2.8. Can you run the following commands on the controller (the machine your run `ansible` from) to see what the output is?
```
$ grep 'equalto' /usr/lib/python2.7/site-packages/jinja2/tests.py
    'equalto':          operator.eq,

$ python -c 'from jinja2 import __version__; print(__version__)'
2.10.1
```

needs_info

**Comment by mkrizek** [comment]
A few related issues https://github.com/ansible/ansible/issues/49166, https://github.com/ansible/ansible/issues/31762, https://github.com/ansible/ansible/issues/10123.

**Comment by sylviapossenau** [comment]
```
grep 'equalto' command returns nothing and 
$ python -c 'from jinja2 import __version__; print(__version__)'
2.7.2
```
The linked issues above all suggest upgrading jinja2 via pip, which I did before opening this issue. Pip installed version 2.10.1 of Jinja2, but as the python command suggests, there seems to be a 2.7.2 version somewhere on my system that gets precedence over the newer version.

Since I did not change any ansible configs, I guess my question is, what do I need to do that this newer version gets picked up?
